### ISSUE TITLE
Methods not callable from outside ipython

### ISSUE THREAD

**Issue by TcaManager** [cfelder/ldap3-orm#2]
Hi, I've created an EntryType with this code
```python
from ldap3_orm import EntryType
from config.connection import CONNECTION,BASE_DN

proxirhUser = EntryType(
    'uid={uid},ou=people,%s' % (BASE_DN),
    ['person','organizationalPerson','inetOrgPerson','educationnationale'], 
    CONNECTION
)
```
Works perfectly, I can add thousands of users like this with the add function.
When I create an instance of proxiUser in a script, it has several methods that I can't use, is there a way to avoid that? I can't use the __entry_to_ldif()__ for example
```python
user1 = proxiUser(**data) # data is a dict, not important
print(user1.entry_to_ldif())
```
Raises the error
```python
Traceback (most recent call last):

  File "<ipython-input-24-8a6fd88903be>", line 1, in <module>
    ldap_user0.entry_to_ldif()

  File "/home/julien/anaconda3/lib/python3.7/site-packages/ldap3/abstract/entry.py", line 339, in entry_to_ldif
    ldif_lines = operation_to_ldif('searchResponse', [self._state.response], all_base64, sort_order=sort_order)

  File "/home/julien/anaconda3/lib/python3.7/site-packages/ldap3/protocol/rfc2849.py", line 209, in operation_to_ldif
    lines = search_response_to_ldif(entries, all_base64, sort_order)

  File "/home/julien/anaconda3/lib/python3.7/site-packages/ldap3/protocol/rfc2849.py", line 127, in search_response_to_ldif
    if 'dn' in entry:

TypeError: argument of type 'NoneType' is not iterable
```

**Comment by cfelder** [comment]
Thanks for reaching out @TcaManager. Unfortunately ``entry_to_ldif`` respectively ``operation_to_ldif`` relies on having a response in its ``EntryState`` instance ``self._state`` which isn't the case when creating new objects without sending a request to the server. Imho this seems to be an upstream issue because building new objects in ldap3 is supported using ldap3's Writer cursors but getting ldif output using ``entry_to_ldif`` fails here.

The following pure ldap3 example fails:
```python
from ldap3 import ObjectDef, Writer
from ldap3.utils.dn import safe_dn

from ldap3_orm.config import config
from ldap3_orm.connection import conn


w = Writer(conn, ObjectDef('inetOrgPerson', conn))
w.uid = 'john'
w.givenName = 'John'
w.sn = 'Doe'
w.cn = 'John Doe'
entry = w.new(safe_dn(('uid={uid},ou=People,' + config.base_dn).format(uid=w.uid)))
entry.entry_to_ldif()  # fails
```

I will report this issue upstream after some further investigation.
Meanwhile as a workaround you can use a _fake_ connection to generate ldif output without sending requests to a server. This is used in the ldap3 test suite as well.

See my minimal working example below:

```python
from ldap3 import LDIF
from ldap3_orm import Connection, EntryType
from ldap3_orm.config import config


Person = EntryType('uid={uid},ou=People,' + config.base_dn, 'inetOrgPerson', conn)
john = Person(uid='john', givenName='John', sn='Doe', cn='John Doe')
ldifconn = Connection(server=None, client_strategy=LDIF)
ldif = ldifconn.add(john.entry_dn, john.object_classes, john.entry_attributes_as_dict)
print(ldif)
```

This generates the following output:

```
version: 1
dn: uid=john,ou=People,dc=example,dc=com
changetype: add
objectClass: inetOrgPerson
uid: john
givenName: John
sn: Doe
cn: John Doe

**Comment by cfelder** [comment]
The following issue is already facing a similar problem using reader cursors.
cannatag/ldap3#689

**Comment by cfelder** [comment]
> Works perfectly, I can add thousands of users like this with the add function.
When I create an instance of proxiUser in a script, it has several methods that I can't use, is there a way to avoid that? I can't use the entry_to_ldif() for example

@TcaManager I've just released 2.5.1 which should mitigate the issue.
For using `entry_to_ldif` I am looking forward to get a response from upstream. Meanwhile you can use the workaround mentioned above.

Best
Christian

**Comment by TcaManager** [comment]
Hi, The virtual connection is very very useful in my case, thanks a lot, it saves me a lot of trouble.

**Comment by TcaManager** [comment]
One question, would it work with modify or delete?
```
ldifconn = Connection(server=None, client_strategy=LDIF)
ldif = ldifconn.modify(john.entry_dn, john.object_classes, john.entry_attributes_as_dict)
```

**Comment by TcaManager** [comment]
Another question, I have something strange in the ldif output, truncated after 78 chars, like this
```
version: 1
dn: uid=atheblues,ou=personnels EN,ou=ac-aix-marseille,ou=education,o=gouv,c=f
 r
```
I created an entry type with the following parameters
```
BASE_DN = 'ou=education,o=gouv,c=fr'
USER_DN = 'uid={uid},ou=personnels EN,ou={FrEduAca}'
USER_OBJECT_CLASSES = ['top','person','organizationalPerson','inetOrgPerson','educationnationale','FrEduFederation']
DN = '%s,%s' % (USER_DN,BASE_DN)
user = EntryType(
    DN,
    USER_OBJECT_CLASSES, 
    connection
)
```
And then an user with
```
agathe = {
    'uid':'atheblues',
    'FrEduAca': 'ac-aix-marseille',
    'givenName':'Agathe',
    'sn':'The Blues',
    'cn':'Agathe The Blues',
    'userPassword':hashed(HASHED_SALTED_SHA, 'OhYeah!'),
    'FrEduRneResp': ['eeeee','fff']
}
```
Ì get this output
```
version: 1
dn: uid=atheblues,ou=personnels EN,ou=ac-aix-marseille,ou=education,o=gouv,c=f
 r
changetype: add
objectClass: top
objectClass: person
objectClass: organizationalPerson
objectClass: inetOrgPerson
objectClass: educationnationale
objectClass: FrEduFederation
uid: atheblues
FrEduAca: ac-aix-marseille
givenName: Agathe
sn: The Blues
cn: Agathe The Blues
userPassword: {ssha}HPJq6yzyzoKCwVAPsKKmBeif3p4JaVSqBOtxkA==
FrEduRneResp: eeeee
FrEduRneResp: fff
```
This other user, with a shorter dn, did not get truncated
```
version: 1
dn: uid=bleclown,ou=personnels EN,ou=ac-caen,ou=education,o=gouv,c=fr
changetype: add
objectClass: top
objectClass: person
objectClass: organizationalPerson
objectClass: inetOrgPerson
objectClass: educationnationale
objectClass: FrEduFederation
uid: bleclown
FrEduAca: ac-caen
givenName: Bozo
sn: Le Clown
cn: Bozo Le Clown
userPassword: {ssha}SfffTwMEJN0vjp6mxNcCjwEBFFJrVWqYgw3H4w==
FrEduRneResp: aaaaa
FrEduRneResp: bbbbbb
```

**Comment by TcaManager** [comment]
Ok, sorry, I didn't know it was the standard behavior of ldif files and ldap tools.

**Comment by cfelder** [comment]
I am closing this as there is a upstream fix in cannatag/ldap3#689 included v2.6.1.
Using the virtual LDIF connection it is also possible to create ldid output as mentioned in this ticket. Unfortunately it's still not possible to use ``entry_to_ldif`` on fresh entry instances but this is an upstream issue.

Feel free to reopen if I missed something here.
